### PR TITLE
Improve Cosmos with local emulator

### LINKED ISSUE #2006: Add support for incoming Cosmos client change for ignoring certificate on connection string
The Cosmos team is making a change to the Cosmos client to make emulator usage easier by ignoring the emulator certificate. Once this is done we can remove some workarounds we added to the Aspire Cosmos components. Tracking in P3 but we won't hold P3 for it if it doesn't make it.

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/aspire/issues/2006
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2008)

### CODE REVIEW

File: src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs [hunk 1]
```diff
@@ -93,12 +93,8 @@ public static class AspireAzureCosmosDBExtensions
...
 
-        if (settings.IgnoreEmulatorCertificate && CosmosUtils.IsEmulatorConnectionString(settings.ConnectionString))
+        if (CosmosUtils.IsEmulatorConnectionString(settings.ConnectionString))
         {
-            clientOptions.HttpClientFactory = () => new HttpClient(new HttpClientHandler()
-            {
-                ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
-            });
             clientOptions.ConnectionMode = ConnectionMode.Gateway;
             clientOptions.LimitToEndpoint = true;
```
Comment (JamesNK) [inline src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs:99]: Should these options be set when talking to the emulator? I tried commenting them out and Cosmos sample failed to work.

Comment (mitchdenny) [reply src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs:99]: My understanding is that the incoming change from the Cosmos team will do this for us. Some get to remove this code.

**Review by mitchdenny — APPROVED** [review]
Lgtm, we just need to get our hands on the new cosmos package now.

File: src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs (hunk 1 above)
Comment (kirankumarkolli) [reply src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs:99]: Both mode and LimitToEndpoint can be removed. 

Documentation was stale and needs to refresh. Is the cause for both gateway-mode and limit
https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-develop-emulator?tabs=windows%2Ccsharp&pivots=api-nosql

[ServerCertificateCustomValidationCallback](https://github.com/Azure/azure-cosmos-dotnet-v3/commit/c29abfedcbd98edf5ac49c9e58c3befc21dd9c3e) do cover for both gateway and direct modes.

Comment (JamesNK) [reply src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs:99]: I've tried removing both and it breaks. I've tried removing them individually and it breaks.

I'm not going to remove either.

**Pull request merged**